client/app/pages/alert/Alert.jsx

```diff
@@ -229,47 +241,51 @@ class AlertPage extends React.Component {
     };
 
     return (
-      <div className="container alert-page">
-        {mode === MODES.NEW && <AlertNew {...commonProps} />}
-        {mode === MODES.VIEW && (
-          <AlertView canEdit={canEdit} onEdit={this.edit} muted={muted} unmute={this.unmute} {...commonProps} />
-        )}
-        {mode === MODES.EDIT && <AlertEdit cancel={this.cancel} {...commonProps} />}
+      <div className="alert-page">
+        <div className="container">
+          {mode === MODES.NEW && <AlertNew {...commonProps} />}
+          {mode === MODES.VIEW && (
+            <AlertView canEdit={canEdit} onEdit={this.edit} muted={muted} unmute={this.unmute} {...commonProps} />
+          )}
+          {mode === MODES.EDIT && <AlertEdit cancel={this.cancel} {...commonProps} />}
+        </div>
       </div>
     );
   }
 }
 
-export default function init(ngModule) {
-  ngModule.component("alertPage", react2angular(AlertPage));
-
-  return routesToAngularRoutes(
-    [
-      {
-        path: "/alerts/new",
-        title: "New Alert",
-        mode: MODES.NEW,
-      },
-      {
-        path: "/alerts/:alertId",
-        title: "Alert",
-        mode: MODES.VIEW,
-      },
-      {
-        path: "/alerts/:alertId/edit",
-        title: "Alert",
-        mode: MODES.EDIT,
-      },
-    ],
-    {
-      template: "<alert-page></alert-page>",
-      controller($scope, $exceptionHandler) {
-        "ngInject";
-
-        $scope.handleError = $exceptionHandler;
-      },
-    }
-  );
-}
-
-init.init = true;
+export default [
+  {
+    path: "/alerts/new",
+    title: "New Alert",
+    render: currentRoute => (
+      <AuthenticatedPageWrapper key={currentRoute.key}>
+        <ErrorBoundaryContext.Consumer>
+          {({ handleError }) => <AlertPage {...currentRoute.routeParams} mode={MODES.NEW} onError={handleError} />}
+        </ErrorBoundaryContext.Consumer>
+      </AuthenticatedPageWrapper>
+    ),
+  },
+  {
+    path: "/alerts/:alertId([0-9]+)",
+    title: "Alert",
+    render: currentRoute => (
+      <AuthenticatedPageWrapper key={currentRoute.key}>
+        <ErrorBoundaryContext.Consumer>
+          {({ handleError }) => <AlertPage {...currentRoute.routeParams} mode={MODES.VIEW} onError={handleError} />}
+        </ErrorBoundaryContext.Consumer>
+      </AuthenticatedPageWrapper>
+    ),
+  },
+  {
+    path: "/alerts/:alertId([0-9]+)/edit",
+    title: "Alert",
+    render: currentRoute => (
+      <AuthenticatedPageWrapper key={currentRoute.key}>
```

Comment:
Why do we need to duplicate this piece of logic (AuthenticatedPage + ErrorBoundry) on every page?

Reply:
Also, passing `handleError` to every page feels like too much. Can't we solve this with some global state/context?

Reply:
As I mentioned in reply to @gabrieldutra, I'm going to revisit this and few other things in a follow-up PR; currently it's stable, and I prefer not to add any more changes to this PR.

Reply:
> Also, passing handleError to every page feels like too much. Can't we solve this with some global state/context?

Yes, and that's what I plan to do. There is a context already though; it's easy to use it with function-based components and a bit trickier with class-based ones - I didn't want to deal with it now so just passed it as a prop everywhere.